d2l-tspan.html

```diff
@@ -1,78 +1,87 @@
-		<d2l-td id="cell">
-			<div id="float">
-				<resize-aware on-element-size-changed="_onSizeChanged">
-					<slot></slot>
-				</resize-aware>
-			</div>
-		</d2l-td>
-	</template>
-	<script>
-		Polymer({
-			is: 'd2l-tspan',
-			behaviors: [
-				D2L.PolymerBehaviors.Table.LocalObserverBehavior
-			],
-			properties: {
-				role: {
-					type: String,
-					value: 'row',
-					readonly: true,
-					reflectToAttribute: true
-				},
-				selected: {
-					type: Boolean,
-					value: false,
-					reflectToAttribute: true,
-					observer: '_selectedChanged'
-				}
-			},
-			_selectedChanged: function(selected) {
-				this.setAttribute('aria-selected', selected.toString());
-			},
-			_onSizeChanged: function() {
-				if (this.root.host) {
-					this.root.host.style.height = this.$.float.offsetHeight + 'px';
-				}
-			}
-		});
-	</script>
-</dom-module>
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="d2l-table-observer-behavior.html">
+<link rel="import" href="../d2l-resize-observer-polyfill-import/resize-observer.html">
+<link rel="import" href="d2l-td.html">
+
+<dom-module id="d2l-tspan">
+	<template strip-whitespace>
+		<style>
+			:host {
+				margin-left: 0;
+				display: table-row;
+			}
+
+			#cell {
+				padding: 0;
+				border: none;
+			}
+
+			#float {
+				position: absolute;
+				width: 100%;
+				border: var(--d2l-table-border);
+				border-bottom: none;
+				padding: 1rem;
+				left: 0;
+				box-sizing: border-box;
+			}
+
+			#float.d2l-table-row-first {
+				border-top-left-radius: var(--d2l-table-border-radius);
+				border-top-right-radius: var(--d2l-table-border-radius);
+			}
+
+			#float.d2l-table-row-last {
+				border-bottom: var(--d2l-table-border);
+				border-bottom-left-radius: var(--d2l-table-border-radius);
+				border-bottom-right-radius: var(--d2l-table-border-radius);
+			}
+		</style>
+		<d2l-td id="cell">
+			<div id="float">
+				<slot></slot>
+			</div>
+		</d2l-td>
+	</template>
+	<script>
+		Polymer({
+			is: 'd2l-tspan',
+			behaviors: [
+				D2L.PolymerBehaviors.Table.LocalObserverBehavior
+			],
+			properties: {
+				role: {
+					type: String,
+					value: 'row',
+					readonly: true,
+					reflectToAttribute: true
+				},
+				selected: {
+					type: Boolean,
+					value: false,
+					reflectToAttribute: true,
+					observer: '_selectedChanged'
+				}
+			},
+			ready: function() {
+				this._sizeObserver = new ResizeObserver(this._onSizeChanged.bind(this));
+			},
+			attached: function() {
+				var slot = Polymer.dom(this.root).querySelector('#float');
+				this._sizeObserver.observe(slot);
+			},
+			detached: function() {
+				var slot = Polymer.dom(this.root).querySelector('#float');
+				this._sizeObserver.unobserve(slot);
+			},
+			_selectedChanged: function(selected) {
+				this.setAttribute('aria-selected', selected.toString());
+			},
+			_onSizeChanged: function() {
+				if (this.root.host) {
+					this.root.host.style.height = this.$.float.offsetHeight + 'px';
```

Comment:
I am guessing we are unable to use the `contentRect` on the entries passed to this observer?  Ideally we can use the measurements provided by the resize observer and avoid the measurement here.

Reply:
Cool. I'm not that familiar with the observer. Will look into it

Reply:
Are you confused or sad?

Reply:
Haha, sad.  Frick'n emoji confusion.  Lol.